py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py

```diff
@@ -532,7 +548,10 @@ def run(
         # self.engine_cache could be None if:
```

Comment:
We are kind of avoiding this but I feel like we might want to restructure completely `run` to assume refit and check if the user wants immutable weights. Also we might want to pull the cache pulling and inserting code into helpers just to make easier to understand

Reply:
Like lines from 551-598 and 671-711 should probably be helpers that pull and insert the weight stripped engine

Reply:
Something like 

```py
if self.reuse_cached_engines:
    def _pull_cached_engine(hash, settings, inputs, 
```